echo-rest/src/main/groovy/com/netflix/spinnaker/echo/events/RestEventListener.groovy

```diff
@@ -70,7 +77,10 @@ class RestEventListener implements EchoEventListener {
           eventAsMap.details = mapper.writeValueAsString(eventAsMap.details)
         }
 
-        if (service.config.wrap) {
+        if (restEventParser.isPresent()) {
+          RestEventParser.Event e = mapper.convertValue(event, RestEventParser.Event)
+          sentEvent = restEventParser.get().parse(e)
+        } else if (service.config.wrap) {
```

Comment:
think that there would be a case to have the eventParser process the event but still want the flatten or wrap behavior to happen on the processed event?

Thinking of a half-baked example of an eventParser that augments all the events with something like an instanceId that handled the event but otherwise they fan out to all the registered services as normal.

Maybe that is less of a concern if we get to the service.config specifying specific eventParsers to apply and make the final form totally up to the eventParser.

I could honestly go either way here..

Reply:
I don't think `flatten` will work with a custom parser (since flatten requires `content` and `details` fields and a custom parser could parse the event to any schema).  However, I can make this work with `wrap`.